src/Plutarch/Extra/List.hs

```diff
@@ -209,3 +238,132 @@ pfromList ::
   [Term s a] ->
   Term s (list a)
 pfromList = foldr (\x xs -> pcons # x # xs) pnil
+
+{- | Check if two lists are equal, given a custom equalator. The two list types
+     can be different.
+
+     @since 3.14.1
+-}
+plistEqualsBy ::
+  forall
+    (list1 :: (S -> Type) -> (S -> Type))
+    (list2 :: (S -> Type) -> (S -> Type))
+    (a :: S -> Type)
+    (b :: S -> Type)
+    (s :: S).
+  (PIsListLike list1 a, PIsListLike list2 b) =>
+  Term s ((a :--> b :--> PBool) :--> list1 a :--> list2 b :--> PBool)
+plistEqualsBy = phoistAcyclic $
+  plam $ \eq -> pfix #$ plam $ \self l1 l2 ->
+    pelimList
+      ( \x xs ->
+          pelimList
+            ( \y ys ->
+                -- Avoid comparison if two lists have different length.
+                self # xs # ys #&& eq # x # y
+            )
+            -- l2 is empty, but l1 is not.
+            (pconstant False)
+            l2
+      )
+      -- l1 is empty, so l2 should be empty as well.
+      (pnull # l2)
+      l1
+
+{- | / O(n) /. Remove the first occurance of a value from the given list. Return
+     nothing is said value is not in the list.
+
+     @since 3.14.1
+-}
+pdeleteBy ::
```

Comment:
We should put first in the name to be clear. Perhaps, `pdeleteFirstBy`

Reply:
Good idea.